### PR TITLE
Add support for Azure, OpenAI, Palm, Anthropic, Cohere Models - using litellm

### PR DESCRIPTION
**Pull request by ishaan-jaff**
I'm the maintainer of litellm https://github.com/BerriAI/litellm - a simple & light package to call OpenAI, Azure, Cohere, Anthropic, Replicate API Endpoints

This PR adds support for models from all the above mentioned providers, through `def make_litellm_chat_request()`
Here's a sample of how litellm is used:

```
from litellm import completion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
@chrislemke if this looks good, I'll add more comprehensive tests using litellm and update docs 😊

**Comment by chrislemke** [comment]
Hey, @ishaan-jaff! 
Thanks for contributing. Looks great so far!!

**Comment by ishaan-jaff** [comment]
@chrislemke added some info to the ReadMe and made some code fixes. 

Any other suggested changes before merging ?

File: workflow/src/text_chat.py [hunk 1]
```diff
@@ -287,8 +290,72 @@ def make_chat_request(
...
+                "presence_penalty": presence_penalty,
+            },
+        )
+
+    return prompt, response
+
+
 exit_on_error()
-__prompt, __response = make_chat_request(
+__prompt, __response = make_litellm_chat_request(
```
Comment (chrislemke) [inline workflow/src/text_chat.py:358]: Hey! Nice job! Just wondering. How is the `model (str)` parameter passed to the `make_litellm_chat_request` function?

**Review by chrislemke — CHANGES_REQUESTED** [review]

File: workflow/src/text_chat.py (hunk 1 above)
Comment (elsatch) [reply workflow/src/text_chat.py:441]: Checking the PR by @ishaan-jaff model(str) should be one of the supported models at https://litellm.readthedocs.io/en/latest/supported/

In the current configuration, there is a dropdown model called ChatGPT model that has 'ChatGPT', 'GPT-4' and 'GPT-4 (32k)'. IMHO, this dropdown could be renamed 'GPT Model' (more recognizable than LLM Model) and include the strings for the supported models.

To me the transition as seamless as possible.

**Pull request merged**